.github/workflows/publish_public_package.yaml

```
@@ -61,56 +56,48 @@ jobs:
-          $ApiKey = az keyvault secret show --id '${{ steps.key-vault-info.outputs.KeyVaultUrl }}/secrets/ScubaGear-PSGAllery-API-Key' --query value -o tsv
-          if (-Not $ApiKey)
-          {
-            Write-Error "Failed to retrieve API key"
-          }
-          # Setup the parameters
-          $Parameters = @{
-            AzureKeyVaultUrl = '${{ steps.key-vault-info.outputs.KeyVaultUrl }}'
-            CertificateName = '${{ steps.key-vault-info.outputs.KeyVaultCertificateName }}'
-            ModuleSourcePath = 'repo/PowerShell/ScubaGear'
-            GalleryName = 'PSGallery'
-            NuGetApiKey = $ApiKey
-          }
-          if ('true' -eq '${{ inputs.IsPrerelease }}')
-          {
-            Write-Output "Adding IsPrerelease"
-            Write-Output ${{ inputs.IsPrerelease }}
-            $Parameters.Add('PrereleaseTag', '${{ inputs.PrereleaseTag }}')
-          }
-          if (-Not [string]::IsNullOrEmpty('${{ inputs.OverrideModuleVersion }}'))
-          {
-            Write-Output "Adding OverrideModuleVersion"
-            Write-Output ${{ inputs.OverrideModuleVersion }}
-            $Parameters.Add('OverrideModuleVersion', '${{ inputs.OverrideModuleVersion }}')
-          }
+          $ApiKey = Get-PSGalleryApiKey -KeyVaultUrl '${{ steps.key-vault-info.outputs.KeyVaultUrl }}'
+
+          # Setup the parameters using the new function
+          $Parameters = Set-PublishParameters `
+            -AzureKeyVaultUrl '${{ steps.key-vault-info.outputs.KeyVaultUrl }}' `
+            -CertificateName '${{ steps.key-vault-info.outputs.KeyVaultCertificateName }}' `
+            -ModuleSourcePath 'repo/PowerShell/ScubaGear' `
+            -ApiKey $ApiKey `
+            -IsPrerelease ('true' -eq '${{ inputs.IsPrerelease }}') `
+            -PrereleaseTag '${{ inputs.PrereleaseTag }}' `
+            -OverrideModuleVersion '${{ inputs.OverrideModuleVersion }}'
+
           # This publishes to PSGallery.
           Publish-ScubaGearModule @Parameters
       # This is a manual test that simply writes the version to the console
       - name: Test Scuba Version
         run: |
-          # Give PSGallery a chance to publish the pre-release to its API
-          Start-Sleep -Seconds 30
-          if ('true' -eq '${{ inputs.IsPrerelease }}')
-          {
-            $Version = '${{ inputs.OverrideModuleVersion }}' + '-' + '${{ inputs.PrereleaseTag }}'
-            Write-Output "Checking for prerelease with required version: $Version"
-            Find-Module -Name ScubaGear -RequiredVersion $Version -AllowPrerelease
-          }
-          else
-          {
-            Write-Output "Installing latest version"
-            Find-Module -Name ScubaGear
+          . repo/utils/workflow/Publish-PublicPackage.ps1
+
+          # Test that the module was published successfully
+          $TestResult = Test-PublishedModule `
+            -IsPrerelease ('true' -eq '${{ inputs.IsPrerelease }}') `
```

Same here, just move the condition into the Test-PublishedModule function.

```suggestion
            -IsPrerelease '${{ inputs.IsPrerelease }}' `
```